horizon/enforcer/utils/statistics_utils.py

```diff
@@ -0,0 +1,47 @@
+import asyncio
+
+
+class StatisticsManager:
+    def __init__(self, interval_seconds: int = 2 * 60):
+        self._requests = 0
+        self._failures = 0
+        self._messages: asyncio.Queue[bool] = asyncio.Queue()
+        self._lock = asyncio.Lock()
+        self._restarter_task = None
+        self._interval_task = None
+        self._interval_seconds = interval_seconds
+
+    async def restarter_task(self) -> None:
+        while True:
+            message = await self._messages.get()
+            try:
+                async with self._lock:
+                    self._requests += 1
+                    if message is False:
+                        self._failures += 1
+            finally:
+                self._messages.task_done()
+            await asyncio.sleep(0.5)
```

Comment:
why do you need to sleep here?
won't we await without blocking on the next message in the queue?